tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/httpclient/Partition.java

```diff
@@ -0,0 +1,29 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+
+package com.linagora.tmail.mailet.rag.httpclient;
+
+public record Partition(String partitionName) {
+
+    public static Partition fromPattern(String pattern, String localPart, String domainName) {
+        String partitionName = pattern
+            .replace("{localPart}", localPart)
+            .replace("{domainName}", domainName);
```

Comment:
What happens if the configured `ragondin.partition.pattern` does not contain `{localPart}` or `{domainName}`?

Maybe we should enforce their existence when initialize the `RagConfig`?